### PR TITLE
tls: refactor tls tests

### PR DESCRIPTION
**Pull request by ramaraochavali**
*Description*: One of the review suggestions in #4810 is to move some of the variable initialization to class level. Here is the [related discussion](https://github.com/envoyproxy/envoy/pull/4810#discussion_r227428951) . This PR addresses that feedback.
*Risk Level*: Low
*Testing*: Existing automated tests
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@lizan @jmarantz fyi. Can you PTAL?

File: test/common/ssl/context_impl_test.cc [hunk 1]
```diff
@@ -279,24 +269,21 @@ TEST_F(SslContextImplTest, TestGetCertInformationWithExpiration) {
...
   EXPECT_EQ(nullptr, context->getCaCertInformation());
   EXPECT_EQ(nullptr, context->getCertChainInformation());
 }
 
 class SslServerContextImplTicketTest : public SslContextImplTest {
 public:
-  static void loadConfig(ServerContextConfigImpl& cfg, Event::TestTimeSystem& time_system) {
-    ContextManagerImpl manager(time_system);
-    Stats::IsolatedStoreImpl store;
+  static void loadConfig(ServerContextConfigImpl& cfg, ContextManagerImpl& manager,
```
Comment (jmarantz) [inline test/common/ssl/context_impl_test.cc]: why not make this non-static so manager and store can be ref'd from class?

File: test/common/ssl/context_impl_test.cc [hunk 2]
```diff
@@ -279,24 +269,21 @@ TEST_F(SslContextImplTest, TestGetCertInformationWithExpiration) {
...
+                         Stats::IsolatedStoreImpl& store) {
     ServerContextSharedPtr server_ctx(
         manager.createSslServerContext(store, cfg, std::vector<std::string>{}));
   }
 
   static void loadConfigV2(envoy::api::v2::auth::DownstreamTlsContext& cfg,
```
Comment (jmarantz) [inline test/common/ssl/context_impl_test.cc]: ditto...also time_system_

File: test/common/ssl/context_impl_test.cc [hunk 3]
```diff
@@ -307,15 +294,16 @@ class SslServerContextImplTicketTest : public SslContextImplTest {
 
     NiceMock<Server::Configuration::MockTransportSocketFactoryContext> factory_context;
     ServerContextConfigImpl server_context_config(cfg, factory_context);
-    loadConfig(server_context_config, time_system);
+    loadConfig(server_context_config, manager, store);
   }
 
-  static void loadConfigYaml(const std::string& yaml, Event::TestTimeSystem& time_system) {
+  static void loadConfigYaml(const std::string& yaml, ContextManagerImpl& manager,
```
Comment (jmarantz) [inline test/common/ssl/context_impl_test.cc]: same here.

File: test/common/ssl/context_impl_test.cc (hunk 1 above)
Comment (ramaraochavali) [reply test/common/ssl/context_impl_test.cc]: Yes. Agree. We can change it like that.

**Comment by ramaraochavali** [comment]
@jmarantz made the methods non-static. PTAL.

**Pull request merged**